src/main/scala/tech/cryptonomic/conseil/tezos/DatabaseConversions.scala

```diff
@@ -125,15 +126,34 @@ object DatabaseConversions extends LazyLogging {
     }
 
   implicit val blockAccountsToAccountHistoryRows =
-    new Conversion[List, BlockTagged[Map[AccountId, Account]], Tables.AccountsHistoryRow] {
-      override def convert(from: BlockTagged[Map[AccountId, Account]]): List[Tables.AccountsHistoryRow] =
-        blockAccountsToAccountRows.convert(from).map {
-          _.into[Tables.AccountsHistoryRow]
-            .withFieldConst(_.asof, Timestamp.from(from.timestamp.getOrElse(Instant.ofEpochMilli(0))))
-            .withFieldConst(_.cycle, from.cycle)
-            .withFieldConst(_.isBakerDeactivated, None)
+    new Conversion[List, (BlockTagged[Map[AccountId, Account]], List[AccountsHistoryRow]), Tables.AccountsHistoryRow] {
+      override def convert(
+          from: (BlockTagged[Map[AccountId, Account]], List[AccountsHistoryRow])
+      ): List[Tables.AccountsHistoryRow] = {
+        val (blockTaggedAccounts, accountsHistoryRowsWithInactiveBakers) = from
+        val touchedAccounts = blockAccountsToAccountRows.convert(blockTaggedAccounts).map { accountsRow =>
+          val isActiveBaker =
+            if (accountsHistoryRowsWithInactiveBakers.map(_.accountId).contains(accountsRow.accountId))
+              Some(false)
+            else if (accountsRow.isBaker)
+              Some(true)
+            else
+              None
+
+          accountsRow
+            .into[Tables.AccountsHistoryRow]
+            .withFieldConst(_.asof, Timestamp.from(blockTaggedAccounts.timestamp.getOrElse(Instant.ofEpochMilli(0))))
+            .withFieldConst(_.cycle, blockTaggedAccounts.cycle)
+            .withFieldConst(_.isActiveBaker, isActiveBaker)
             .transform
         }
+        val notTouchedAccounts =
+          accountsHistoryRowsWithInactiveBakers
+            .map(_.copy(isActiveBaker = Some(false)))
+            .filterNot(accountsHistoryRow => touchedAccounts.map(_.accountId).contains(accountsHistoryRow.accountId))
```

Comment:
we should avoid remapping all touched accounts for the id, in each loop cycle... better to keep a local set with the 
`val touched = touchedAccounts.map(_.accountId).toSet`, and reuse it for each row entry

Reply:
done